plugins/http_plugin/include/eosio/http_plugin/beast_http_session.hpp

```diff
@@ -150,10 +163,12 @@ class beast_http_session : public detail::abstract_conn {
             if(plugin_state_->logger.is_enabled(fc::log_level::all))
                plugin_state_->logger.log(FC_LOG_MESSAGE(all, "resource: ${ep}", ("ep", resource)));
             std::string body = req.body();
-            handler_itr->second(derived().shared_from_this(),
+            auto content_type = handler_itr->second.content_type;
```

Comment:
I would expect you to parse the content type from the request header and send response based on the parsed content type. However, you didn't seem to do that.

Reply:
We can add that.  But, currently the code always returns json.  Probably a lot of api clients specify */*.  So we have to fall back to something.  The content type I added for api is the something.  I really don't see the use case for returning a json encoded prometheus string, though.

Reply:
You need to keep two different maps for handlers, like `url_json_handlers` and `url_text_handlers`, and when you need to call the handler, you provide different http response handler to the user provided handler. For example, if the request content type is json, you find the handler from url_json_handlers and call the handler with  
 the result of `make_http_json_reponse_handler()` and the `http_json_reponse_handler` should do the json encoding.